test/functional/test_framework/socks5.py

```diff
@@ -121,7 +122,8 @@ def handle(self):
             logger.exception("socks5 request handling failed.")
             self.serv.queue.put(e)
         finally:
-            self.conn.close()
+            if not self.serv.keep_alive:
+                self.conn.close()
```

Comment:
Good question, I'm just letting it drop out of scope at the end of the test, but [looking into it more](https://docs.python.org/3/library/threading.html#threading.Thread.daemon) it may actually be staying alive until the entire test runner exits.

Should I add a manual close function?

Reply:
update: no longer needed

Reply:
update update: manually calling `onion_proxy.close()` in the test now with a comment